control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go

```diff
@@ -1632,6 +1632,13 @@ func (r *HostedControlPlaneReconciler) reconcileKubeAPIServer(ctx context.Contex
 	}); err != nil {
 		return fmt.Errorf("failed to reconcile api server deployment: %w", err)
 	}
+
+	// Ensure all replicas are updated and ready
+	// https://kubernetes.io/releases/version-skew-policy/#supported-component-upgrade-order
+	if ready, err := util.IsDeploymentReady(ctx, r, kubeAPIServerDeployment); !ready {
+		return fmt.Errorf("kas deployment not ready: %w", err)
```

Comment:
I don't think we should error here as this is part of the expected flow, let's just return and let reconciliation be triggered by the deployment update events.

Reply:
+1 the exponential backoff due to the error can cause a lot of unnecessary delay.